### ISSUE TITLE
Developer installation of arcade raises ContextException on some intel GPUs under Windows 10

### ISSUE THREAD

**Issue by pushfoo** [pythonarcade/arcade#845]
### tl;dr
For certain intel hardware, using `pyglet==2.0.dev1` for developer installs instead of `pyglet==1.5.15` makes all attempts to use arcade raise  `pyglet.gl.ContextException: Unable to share contexts.` 

### Actual behavior:
[After a developer install from source](https://arcade.academy/installation_from_source.html) on certain Intel hardware, launching any arcade window on any commit after [release tag 2.5.6 (commit hash `3bc531e`)](https://github.com/pythonarcade/arcade/commit/3bc531e2e8527e85455c2c877055fabbdaa47d41) can raise `pyglet.gl.ContextException: Unable to share contexts.`

See later sections for full hardware info.

```
(arcade) C:\Users\pushfoo\Projects\arcade>python -m arcade.gui.examples.show_uilayouts_hud_inventory
Traceback (most recent call last):
  File "C:\Users\pushfoo\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\pushfoo\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\pushfoo\Projects\arcade\arcade\gui\examples\show_uilayouts_hud_inventory.py", line 118, in <module>
    main()
  File "C:\Users\pushfoo\Projects\arcade\arcade\gui\examples\show_uilayouts_hud_inventory.py", line 112, in main
    window = Window(resizable=True)
  File "C:\Users\pushfoo\Projects\arcade\arcade\application.py", line 94, in __init__
    resizable=resizable, config=config, vsync=False, visible=visible)
  File "C:\Users\pushfoo\PycharmProjects\arcade\lib\site-packages\pyglet\window\win32\__init__.py", line 130, in __init__
    super(Win32Window, self).__init__(*args, **kwargs)
  File "C:\Users\pushfoo\PycharmProjects\arcade\lib\site-packages\pyglet\window\__init__.py", line 571, in __init__
    self._create()
  File "C:\Users\pushfoo\Projects\arcade\arcade\application.py", line 505, in _create
    super()._create()
  File "C:\Users\pushfoo\PycharmProjects\arcade\lib\site-packages\pyglet\window\win32\__init__.py", line 276, in _create
    self.context.attach(self.canvas)
  File "C:\Users\pushfoo\PycharmProjects\arcade\lib\site-packages\pyglet\gl\win32.py", line 290, in attach
    super(Win32ARBContext, self).attach(canvas)
  File "C:\Users\pushfoo\PycharmProjects\arcade\lib\site-packages\pyglet\gl\win32.py", line 237, in attach
    raise gl.ContextException('Unable to share contexts.')
pyglet.gl.ContextException: Unable to share contexts.
```
Forcibly replacing pyglet with 1.5.15 (the version installed alongside arcade when installing `arcade==2.5.6` from PyPI) makes demos work as expected.

Installations of arcade from PyPI are not affected.

### Expected behavior:
Arcade demos do not break. 

#### 1. Have this hardware & software:

OS: Windows 10
GPU info:
![image](https://user-images.githubusercontent.com/36696816/114878729-8aa7ab00-9dce-11eb-823b-31402be6c53a.png)

I also have a second monitor plugged into the HDMI port as well.

#### 2. Install arcade from source for development per the [developer guide](https://arcade.academy/installation_from_source.html)
```
pip install -e .[dev,docs]
```

#### 3. Run any demo, watch it crash
```
python -m arcade.gui.examples.show_uilayouts_hud_inventory
```

### Additional info

Installing arcade from PyPI installs `pyglet=1.5.15`, whereas a development install from source installs `pyglet==2.0.dev1`.

[Mailing list archives](https://groups.google.com/g/pyglet-users/c/44pS-xkoObI/m/NXyU31cMfPgJ) and[ other github tickets](https://github.com/mmatl/pyrender/issues/13) indicate commonalities among similar issues:
1. Intel integrated GPUs
2. Upgrading pyglet version requirements in modules seem to cause it
3. Turning off shadow windowing can stop this exception

I tried setting pyglet's shadow option at the top of a demo file, but it didn't work. 

Per @einarf 's suggestion in chat, [setting `pyglet.options['shadow_window']` to `False` in arcade/__init__.py for all platforms rather than just Darwin](https://github.com/pythonarcade/arcade/blob/daff5b3ca47138057d0002336a42448bf4f5fed5/arcade/__init__.py#L57-L60) allowed demos to run with `pyglet==2.0.dev1` installed.

**Comment by einarf** [comment]
We should disable shadow window for all platforms to make it simple. Also consider looking at the pyglet config in the initialization. Compare it to the default one and make sure we're doing sane things : https://github.com/pythonarcade/arcade/blob/daff5b3ca47138057d0002336a42448bf4f5fed5/arcade/application.py#L81-L90
